httpx/_config.py

```diff
@@ -410,6 +411,48 @@ def __repr__(self) -> str:
         )
 
 
+class EventHooks(MutableMapping):
```

Comment:
I think if we apply my suggestion of using a plain `Dict[str, List[Callable]]` always, we can mostly drop this custom data structure, correct?

I'm also not sure we _need_ to enforce that the only possible keys be `request` and `response`. Currently we do this with a bit of a surprising behavior ("don't actually store items for unknown keys"), while I think making it so that `event_hooks` behaves just like a normal dict would be beneficial.

So perhaps we could replace this with a simple factory helper…

```python
def create_event_hooks(initial: Dict[str, List[Callable]] = None) -> Dict[str, List[Callable]]):
    event_hooks = {"request": [], "response": []}
    event_hooks.update(initial or {})
    return event_hooks
```